[StyleCop] Fix warnings in Text.Number Project

Bill7zz opened the pull request:
- XML comment analysis warning

SpanishNumberRangeParserConfiguration  …
- Move constructor before properties
- Move constructor initializers on their own line
- Add class access modifier

BasePercentageExtractor  …
- Reorder properties a fields
- Set field to a property to keep the variable name with upper-case
- Add returns comments in methods
- Add trailing coma in multi-line initializers
- Use String.Empty instead empty strings

NumberMode  …
- Update comment to fix CS1570 XML comment has badly formed

Bill7zz committed:
Fix StyleCop warnings in SpanishNumberRangeParserConfiguration
-Move constructor before properties
-Move constructor initializers on their own line
-Add class access modifier

Bill7zz committed:
Fix StyleCop warnings in Text.Number project
-XML comment analysis warning

Aliandi commented on .NET/Microsoft.Recognizers.Text.Number/Extractors/BasePercentageExtractor.cs:
```diff
@@ -8,30 +8,28 @@ namespace Microsoft.Recognizers.Text.Number
-        /// <param name="source"></param>
-        /// <returns></returns>
+        /// <param name="source">sentnce.</param>
+        /// <returns>List of percentaje entities from the sentence source.</returns>
```
change _percentaje_ to _percentage_

Aliandi commented on .NET/Microsoft.Recognizers.Text.Number/Extractors/BasePercentageExtractor.cs:
```diff
@@ -125,13 +124,14 @@ protected static ImmutableHashSet<Regex> BuildRegexes(HashSet<string> regexStrs,
+        /// replace the @sys.num to the real patterns, directly modifies the ExtractResult.
         /// </summary>
-        /// <param name="results">extract results after number extractor</param>
-        /// <param name="originSource">the sentense after replacing the @sys.num, Example: @sys.num %</param>
```
Where did this line go?

Aliandi commented on .NET/Microsoft.Recognizers.Text.Number/Extractors/BasePercentageExtractor.cs:
```diff
@@ -200,14 +200,14 @@ protected static ImmutableHashSet<Regex> BuildRegexes(HashSet<string> regexStrs,
+        /// <param name="numExtResults">number extractor result.</param>
+        /// <returns>return according type "builtin.num" or "builtin.num.percentage".</returns>
+        private string PreprocessStrWithNumberExtracted(
+            string str, out Dictionary<int, int> positionMap, out IList<ExtractResult> numExtResults)
```
Either put them all in the same line like 
` private void PostProcessing(List<ExtractResult> results, string originSource, Dictionary<int, int> positionMap, IList<ExtractResult> numExtResults)`
or one per line
```
 private void PostProcessing(
            List<ExtractResult> results, 
            string originSource, 
            Dictionary<int, int> positionMap,
            IList<ExtractResult> numExtResults)
```

JuanAr commented on .NET/Microsoft.Recognizers.Text.Number/Extractors/BasePercentageExtractor.cs:
```diff
@@ -1,295 +1,301 @@
-﻿using System;
```
This file seems fully updated due to changes in line breaks

Bill7zz replied:
Feedback applied.

JuanAr reviewed (changes requested):
👍  Everything good except the new lines thing in one file

Bill7zz committed:
Fix StyleCop warning in NumberMode
-Update comment to fix CS1570 XML comment on 'variable' has badly formed

Bill7zz committed:
Fix StyleCop warnings in BasePercentageExtractor
-Reorder properties an fields
-Set field to property to keep the variable name with upper-case
-Add retuns comments in methods
-Add trailing coma in multi-line initializers
-Use String.Empty instead empty strings
